include/sound/sof/topology.h

```diff
@@ -163,39 +163,22 @@ struct sof_ipc_comp_tone {
 	int32_t ramp_step;
 } __packed;
 
-/** \brief Types of EFFECT */
-enum sof_ipc_effect_type {
-	SOF_EFFECT_NONE = 0,		/**< None */
-	SOF_EFFECT_INTEL_EQFIR,		/**< Intel FIR */
-	SOF_EFFECT_INTEL_EQIIR,		/**< Intel IIR */
+/** \brief Types of processing components */
+enum sof_ipc_process_type {
+	SOF_PROCESS_NONE = 0,		/**< None */
+	SOF_PROCESS_EQFIR,		/**< Intel FIR */
+	SOF_PROCESS_EQIIR,		/**< Intel IIR */
 };
 
-/* general purpose EFFECT configuration */
-struct sof_ipc_comp_effect {
-	struct sof_ipc_hdr hdr;
-	uint32_t type;			/** sof_ipc_effect_type */
-} __packed;
-
-/* FIR equalizer component */
-struct sof_ipc_comp_eq_fir {
-	struct sof_ipc_comp comp;
-	struct sof_ipc_comp_config config;
-	uint32_t size;
-
-	/* reserved for future use */
-	uint32_t reserved[8];
-
-	unsigned char data[0];
-} __packed;
-
-/* IIR equalizer component */
-struct sof_ipc_comp_eq_iir {
+/* generic "effect", "codec" or proprietary processing component */
+struct sof_ipc_comp_process {
 	struct sof_ipc_comp comp;
 	struct sof_ipc_comp_config config;
-	uint32_t size;
+	uint32_t size;	/**< size of bespoke data section in bytes */
+	uint32_t type;	/**< sof_ipc_effect_type */
```

Comment:
I am also not clear on this one. In the previous PR there was a notion of flavour in the sof_ipc_comp_config, now we have a notion of effect_type. This seems redundant to me. If the goal is to uniquely identify the component we should use a UUID, that's what they were invented.